src/lib/common/sol-platform-linux-common.c

```diff
@@ -263,3 +307,581 @@ sol_platform_impl_get_os_version(char **version)
+    struct sol_str_slice slice = sol_str_slice_from_str(mnt_opts);
+
+    opts = sol_util_str_split(slice, ",", 0);
+    SOL_VECTOR_FOREACH_IDX (&opts, itr, idx) {
+        if (mountopt_str_to_mask(*itr, &flags)) {
+            continue;
+        } else if (sol_str_slice_str_eq(*itr, "noauto")) {
+            *noauto = true;
+            continue;
+        }
+
+        if (!first_custom)
+            sol_buffer_append_char(custom_opts, ',');
+        sol_buffer_append_slice(custom_opts, *itr);
+        first_custom = false;
+
+    }
+    return flags;
+}
+
+static int
+parse_mount_point_file(const char *file, struct sol_ptr_vector *vector)
+{
+    struct mntent mbuf;
+    struct mntent *m;
+    char strings[4096];
+    struct sol_mount_point *mp;
+    uint16_t i;
+    FILE *tab = setmntent(file, "re");
+
+    if (!tab) {
+        if (errno == ENOENT) {
+            SOL_INF("No such %s", file);
+            return -errno;
+        } else {
+            SOL_WRN("Unable to open %s file: %s", file,
+                sol_util_strerrora(-errno));
+            return -errno;
+        }
+    }
+
+    while ((m = getmntent_r(tab, &mbuf, strings, sizeof(strings)))) {
+        struct sol_buffer custom_opts;
+        struct sol_mount_point *mpoint;
+        bool noauto = false;
+
+        mpoint = calloc(1, sizeof(struct sol_mount_point));
+        if (!mpoint) {
+            SOL_ERR("Could not allocate mount point memory");
+            goto err;
+        }
+
+        if (mbuf.mnt_fsname)
+            mpoint->dev = strdup(mbuf.mnt_fsname);
+
+        if (mbuf.mnt_dir)
+            mpoint->mpoint = strdup(mbuf.mnt_dir);
+
+        if (mbuf.mnt_type)
+            mpoint->fstype = strdup(mbuf.mnt_type);
+
+        sol_buffer_init(&custom_opts);
+        mpoint->flags = get_mountflags(mbuf.mnt_opts, &custom_opts, &noauto);
+        mpoint->noauto = noauto;
+
+        if (custom_opts.used)
+            mpoint->opts = sol_buffer_steal(&custom_opts, NULL);
+
+        sol_buffer_fini(&custom_opts);
+
+        if (sol_ptr_vector_append(vector, mpoint) < 0) {
+            SOL_ERR("Could not append mount point to vector");
+            goto err;
+        }
+    }
+    endmntent(tab);
+    return 0;
+
+err:
+    SOL_PTR_VECTOR_FOREACH_IDX (vector, mp, i) {
+        sol_mount_point_free(mp);
+    }
+    sol_ptr_vector_clear(vector);
+    return -ENOMEM;
+}
+
+int
+sol_platform_impl_get_mount_points(bool hplug_only, struct sol_ptr_vector *vector)
+{
+    struct sol_ptr_vector mtab, sol_mtab, delete;
+    struct sol_mount_point *mp, *mp2;
+    uint16_t i, j;
+    int err;
+
+    if (!hplug_only) {
+        return parse_mount_point_file(SYS_MTAB_FILE, vector);
+    }
+
+    sol_ptr_vector_init(&mtab);
+    err = parse_mount_point_file(SYS_MTAB_FILE, &mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_mtab);
+
+    sol_ptr_vector_init(&sol_mtab);
+    err = parse_mount_point_file(SOL_MTAB_FILE, &sol_mtab);
+    SOL_INT_CHECK_GOTO(err, < 0, err_sol_mtab);
+
```

Comment:
Just realized that a generic api to get all the mount points in the system would be fair enough. But, sure, it makes things way much simpler to not consider the system ones and go on only with the hplug mpoints.